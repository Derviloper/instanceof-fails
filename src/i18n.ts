import rawDe from "@/locales/de.json?raw";
import { createI18n } from "vue-i18n";
import { nextTick } from "vue";

const de = JSON.parse(rawDe);

export const SUPPORTED_LOCALES = [
  "de",
  "de-AT",
  "es",
  "fr",
  "fr-LU",
  "it",
  "nl",
  "nl-BE",
  "pl",
];

const i18n = createI18n({
  legacy: false,
  locale: "de",
  datetimeFormats: { de: de.datetimeFormats },
  messages: { de: de.messages },
  numberFormats: { de: de.numberFormats },
});

export default i18n;

export async function loadLocale(locale: string) {
  const config = await import(
    /* webpackChunkName: "locale-[request]" */ `./locales/${locale}.json`
  );
  i18n.global.setDateTimeFormat(locale, config.default.datetimeFormats);
  i18n.global.setLocaleMessage(locale, config.default.messages);
  i18n.global.setNumberFormat(locale, config.default.numberFormats);
  return nextTick();
}

export async function setI18nLanguage(locale: string): Promise<boolean> {
  if (!SUPPORTED_LOCALES.includes(locale)) {
    return false;
  }
  if (!i18n.global.availableLocales.includes(locale)) {
    await loadLocale(locale);
  }
  i18n.global.locale.value = locale;
  /**
   * NOTE:
   * If you need to specify the language setting for headers, such as the `fetch` API, set it here.
   * The following is an example for axios.
   *
   * axios.defaults.headers.common['Accept-Language'] = locale
   */
  document.querySelector("html")?.setAttribute("lang", locale);
  return true;
}
