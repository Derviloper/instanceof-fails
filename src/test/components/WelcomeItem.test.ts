import { describe, expect, it } from "vitest";
import { mount } from "@vue/test-utils";
import WelcomeItem from "@/components/WelcomeItem.vue";
import i18n from "@/i18n";
import router from "@/router";
import FloatingVue from "floating-vue";

describe("WelcomeItem.vue", () => {
  it("should render", () => {
    const wrapper = mount(WelcomeItem, {
      global: { plugins: [i18n, router, FloatingVue] },
      slots: {
        icon: "Icon slot",
        heading: "Heading slot",
        default: "Default slot",
      },
    });
    console.log(wrapper.find("h3"));
    expect(wrapper.html()).toMatchSnapshot();
  });
});
