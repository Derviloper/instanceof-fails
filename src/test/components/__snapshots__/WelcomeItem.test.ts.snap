// Vitest Snapshot v1

exports[`WelcomeItem.vue > should render 1`] = `
"<div xmlns=\\"http://www.w3.org/1999/xhtml\\" class=\\"item\\" data-v-6f9f15d6=\\"\\"><i data-v-6f9f15d6=\\"\\">Icon slot</i>
  <div class=\\"details\\" data-v-6f9f15d6=\\"\\">
    <h3 data-v-6f9f15d6=\\"\\">Heading slot</h3>Default slot
  </div>
</div>"
`;
